glotaran/plugin_system/data_io_registration.py

```diff
@@ -98,15 +101,49 @@ def is_known_data_format(format_name: str) -> bool:
     )
 
 
-def known_data_formats() -> list[str]:
+def known_data_formats(full_names: bool = False) -> list[str]:
     """Names of the registered data io plugins.
 
+    Parameters
+    ----------
+    full_names: bool
+        Whether to display the full names the plugins are
+        registered under as well.
+
     Returns
     -------
     list[str]
         List of registered data io plugins.
     """
-    return registered_plugins(plugin_registry=__PluginRegistry.data_io)
+    return registered_plugins(plugin_registry=__PluginRegistry.data_io, full_names=full_names)
+
+
+def set_data_plugin(
+    format_name: str,
+    full_plugin_name: str,
+) -> None:
+    """Set the plugin used for a specific data format.
+
+    This function is useful when you want to resolve conflicts of installed plugins
+    or overwrite the plugin used for a specific format.
+
+    Effected functions:
+
+    - :func:`load_dataset`
+    - :func:`save_dataset`
+
+    Parameters
+    ----------
+    format_name : str
+        Format to use the plugin for.
```

Comment:
A bit vague what a "format" is. Perhaps refer to the documentation where it is explained in more detail.
In most cases it would refer to a file extension, but what about (future) support for data folders, e.g. a folder containing a specific set of files that together describe a dataset? Doesn't have to be addressed in this PR.

Reply:
How about "Format name to be used with ``save`` and ``load`` functions."? 
Since the argument has the same name as the one in `load_dataset` and `save_dataset`

Reply:
As for a data folder plugin, you could just register it under the name `data_folder` and raise an error in the `save` and `load` methods if `file_name` isn't a folder. The only restriction we have on format names is that they aren't allowed to contain `.`, since this is the quick and dirty way to differentiate between full names and intended names.

Reply:
```rst
    Parameters
    ----------
    format_name : str
        Format name used to refer to the plugin when used for ``save`` and ``load`` functions.
```